free-1.md

```diff
@@ -4,40 +4,120 @@
+style such as
+
+```
+Molecule hydronium = OH-
+Molecule hydroxide = H30+
+Molecule[] products = react(hydronium, hydroxide, conditions)
+products == [(Molecule) H20, (Molecule) H20]
+```
+
+ALternatively, when seeking synthesis pathways (i.e. a known endpoint) recursion 
+works well, such as
+
+```
+Molecule endproduct = <something complicated>
+Molecule startingpoint = synthesize(endproduct)
+```
+
+where `synthesize` is some sort of recursive function.
+
+It meshes well with existing programming language paradigms and designs.
+
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+What excites me about this is that I've been working on something similar since 
+my sophomore year, when I was still a premed student taking Organic Chemistry.
+
+I thought to myself that these rules were relatively straightforward, and it was
+very much an algorithmic process.  I've been working on it off and on since 
+then.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Predicting organic chemistry reaction pathways
+
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+They could use an internal DSL; a library could be written using some preferred
+language and then provide hooks for some end-user to interact with.
+
+They could use an external DSL; a friendlier syntax that provides more 
+constraints on what can be done, and how.
+
+They could use a GUI-based external DSL; they draw molecules and conditions and
+then press a big "react" button.  This is probably the most end-user friendly,
+especially for chemists who aren't great with programming.  
+
+Overall I think each of these should be provided and well linked, if the DSL
+was well funded and supported.  This allows users of many technical backgrounds
+to interact with the software and customize it to meet their needs.  In context 
+to 111, the best is probably internal - this is the easiest to write and test
+because there are fewer layers between the algorithms and the user, and it 
+allows the most technical users to use it intimately and find the pain points.
+
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+It should provide (optionally) verbose output describing what step in the 
+reaction is happening, and how it is determining what step happens.  Errors 
+include invalid molecules (i.e. ones that couldn't possibly exist under those
+conditions), as well as invalid reactions (where nothing would happen there).
+Potentially safety messages as well; i.e. that byproducts are produced that are
+explosive or toxic, or what have you.  These should be exceptions with good 
+informative names and error messages so the user can programmatically handle
+them if they want to.
```

Comment:
Invalid molecules could happen at compile time (if it were a compiled language).  

That seems like it could be argued either way - by throwing an exception you force the user to handle it directly instead of propagating some error down the pipeline because of an unexpected `Null`.  I think I'd like something like `Try<something>` (an idiomatic form in C#).  It would look like

```c#
bool TryParseJsonFile(string jsonFileName, out Dictionary<string, string> jsonData)
{
    try {
        File jsonFile = File.open(jsonFileName, 'r');
    } catch (IOException ex) {
        jsonData = null;
        return false;
    }

    try {
        for {var pair: jsonFile.read()) {
            jsonData.addPair(pair);
        }
    } catch (Exception ex) {
       jsonData = null;
       return false;
    }
    
    return true;  
}
```

which would then be used as

```c#
Dictionary<string, string> dict;
if (TryParseJsonFile(filename, out dict)) {
    // Do something cuz of success
} else {
    // Use some default value
}
```